src/pages/home/report/withReportAndPrivateNotesOrNotFound.tsx

```diff
@@ -1,88 +1,63 @@
-    forwardedRef: PropTypes.func,
-
-    /** The report currently being looked at */
-    report: reportPropTypes,
-
-    /** Information about the network */
-    network: networkPropTypes.isRequired,
-
+type WithReportAndPrivateNotesOrNotFoundOnyxProps = {
     /** Session of currently logged in user */
-    session: PropTypes.shape({
-        /** accountID of currently logged in user */
-        accountID: PropTypes.number,
-    }),
-
-    route: PropTypes.shape({
-        /** Params from the URL path */
-        params: PropTypes.shape({
-            /** reportID and accountID passed via route: /r/:reportID/notes/:accountID */
-            reportID: PropTypes.string,
-            accountID: PropTypes.string,
-        }),
-    }).isRequired,
+    session: OnyxEntry<OnyxTypes.Session>;
 };
 
-const defaultProps = {
-    forwardedRef: () => {},
-    report: {},
-    session: {
-        accountID: null,
-    },
-};
+type WithReportAndPrivateNotesOrNotFoundProps = WithReportAndPrivateNotesOrNotFoundOnyxProps & WithReportOrNotFoundProps & {route: RouteProp<{params: {accountID?: string}}>};
 
-export default function (pageTitle) {
+export default function <TProps extends WithReportAndPrivateNotesOrNotFoundProps, TRef>(pageTitle: TranslationPaths) {
     // eslint-disable-next-line rulesdir/no-negated-variables
-    return (WrappedComponent) => {
+    return (WrappedComponent: ComponentType<TProps & RefAttributes<TRef>>) => {
         // eslint-disable-next-line rulesdir/no-negated-variables
-        function WithReportAndPrivateNotesOrNotFound({forwardedRef, ...props}) {
+        function WithReportAndPrivateNotesOrNotFound(props: TProps & {forwardedRef: ForwardedRef<TRef>}) {
             const {translate} = useLocalize();
-            const {route, report, network, session} = props;
-            const accountID = route.params.accountID;
-            const isPrivateNotesFetchTriggered = !_.isUndefined(report.isLoadingPrivateNotes);
+            const network = useNetwork();
+            const {route, report, session, forwardedRef} = props;
+            const accountID = route.params?.accountID;
+            const isPrivateNotesFetchTriggered = report?.isLoadingPrivateNotes !== undefined;
             const prevIsOffline = usePrevious(network.isOffline);
             const isReconnecting = prevIsOffline && !network.isOffline;
-            const isOtherUserNote = accountID && Number(session.accountID) !== Number(accountID);
+            const isOtherUserNote = accountID && Number(session?.accountID) !== Number(accountID);
             const isPrivateNotesFetchFinished = isPrivateNotesFetchTriggered && !report.isLoadingPrivateNotes;
-            const isPrivateNotesEmpty = accountID ? _.has(lodashGet(report, ['privateNotes', accountID, 'note'], '')) : _.isEmpty(report.privateNotes);
+            const isPrivateNotesEmpty = accountID ? !!report?.privateNotes?.[Number(accountID)].note ?? '' : isEmptyObject(report?.privateNotes);
```

Comment:
```ts
!!report?.privateNotes?.[Number(accountID)].note ?? ''
```

...and what does it mean now? I'm surprised it is accepted by the compiler and linter.

Reply:
Good point, the `?? ''` is not necessary as it allows `isPrivateNotesEmpty` to be a string

Reply:
> as it allows isPrivateNotesEmpty to be a string

I don't think so. That expression should parse as...

```ts
(!!report?.privateNotes?.[Number(accountID)].note) ?? ''
```

...and `!!x` should always be a `boolean`. As I understand it, `!!x ?? y` never evaluates to `y`.

Reply:
Oh now I get what you meant, nvm 😅 I removed `?? ''`

Reply:
@cubuspl42 Maybe this eslint rule would detect this - https://github.com/typescript-eslint/typescript-eslint/blob/main/packages/eslint-plugin/docs/rules/no-unnecessary-condition.md

Reply:
@blazejkustra Do you think we could make use of the mentioned lint?